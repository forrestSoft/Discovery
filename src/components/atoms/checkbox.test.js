import React from 'react'
import {shallow, mount} from 'enzyme'
import shallowToJson from 'enzyme-to-json'

import Checkbox from './checkbox'

it('render matches snapshot', () => {
  const checkbox = shallowToJson(shallow(
  	<Checkbox 
      checked={true}
    />
  )) 
  expect(checkbox).toMatchSnapshot();
});

it('render disabled matches snapshot', () => {
  const checkbox = shallowToJson(shallow(
  	<Checkbox 
  		checked={false}
  	/>
  )) 
  expect(checkbox).toMatchSnapshot();
});

it('calls passed onClick on "onClick"', () => {
	const mock = jest.fn().mockReturnValue('onClick')
	const checkbox = mount(
  	<Checkbox 
  		text={'I am a checkbox'}
  		disabled={false}
  		onChange={mock}
      checked={false}
  	/>
  )

	checkbox.find('input[type="checkbox"]').simulate('change',{ target: { checked: true } })
	expect(mock).toBeCalled()
})

xit('doesnt call passed onClick on "onClick" when disabled', () => {
  // can't really make this happen?
  // so we rely on checking that react renders the disabled attr above
})