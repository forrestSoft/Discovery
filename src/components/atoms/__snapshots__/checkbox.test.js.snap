// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render disabled matches snapshot 1`] = `
<input
  checked={false}
  className="invalid"
  type="checkbox"
/>
`;

exports[`render matches snapshot 1`] = `
<input
  checked={true}
  className="valid"
  type="checkbox"
/>
`;
